### PR TITLE
Completed new shopping list

### LINKED ISSUE #3: 3. As a user, I want to set up a new shopping list so I can start tracking purchased items
## Summary 
A shopping list is a set of items associated with a specific three-word token. We need to allow users to start new lists, so they can save the things they need to buy. To start a new list, we will need to do a few things:

1. Generate a new, unique token
2. Save the token to `localStorage`
3. Show the user the list view

Up to this point, the app has rendered the items in the list associated with the token `my test list`. Before we can allow users to make *new* lists, we have to stop showing them this test list.

## Acceptance criteria
- [x] `@the-collab-lab/shopping-list-utils` is added as a dependency to the project.
- [x] The string `my test list` in `App.jsx` is replaced with `null`, so users are not automatically subscribed to any list.

If a user **doesn’t** already have a token:
- [x] A button in the `Home` component allows them to create a new list
- [x] Clicking the button generates a new token and saves it to `localStorage` using the `setListToken` function in `App.jsx`.
- [x] Once the token has been created and saved, the user is redirected to the `List` view.

If a user **does** already have a token:
- [x] They are automatically redirected to the `List` view.

## Notes
- The  `generateToken` function from the `shopping-list-utils` package should be used to generate the token. More information on that package [its documentation](https://www.npmjs.com/package/@the-collab-lab/shopping-list-utils). Importing it will look something like this:
``` js
import { generateToken } from '@the-collab-lab/shopping-list-utils';
```
- The `setListToken` function takes **one** argument: the new list token.
- Eventually the welcome/home page will look like [this wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png). It’s good for your whole team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add the other features such as joining an existing list.
- For this story, it's sufficient just to have a simple "create list" button. Although you are not required to save the token to Firestore, some teams have opted to also save the token to Firestore when generating a token.

### PR DESCRIPTION
**Pull request by lizzypine**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

- Added @the-collab-lab/shopping-list-utils as a dependency.
- Added a button to Home component.
- Clicking the button generates a new token, which is saved to local storage.
- If the user does not have a token, they are taken to the homepage. Otherwise they are taken to their shopping list.

## Related Issue

Closes #3 

## Acceptance Criteria

 @the-collab-lab/shopping-list-utils is added as a dependency to the project.
- The string my test list in App.jsx is replaced with null, so users are not automatically subscribed to any list.
- If a user doesn’t already have a token:

 - A button in the Home component allows them to create a new list
 - Clicking the button generates a new token and saves it to localStorage using the setListToken function in App.jsx.
 - Once the token has been created and saved, the user is redirected to the List view.
- If a user does already have a token: They are automatically redirected to the List view.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
| ✓ | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
Screenshot 2023-04-10 at 9 44 54 AM

### After

Screenshot 2023-04-10 at 9 43 05 AM

## Testing Steps / QA Criteria
- Make sure you remove any tokens using dev tools => Application => Local storage
- Try going to the homepage. You would expect to see the button with text: "Create a new list."
- After clicking the button, you would expect to be routed to the /list page.

### CODE REVIEW

**Review by 01001101CK — APPROVED** [review]
Looks great to Olivia and me! Would love to learn more about useCallback.

File: src/views/Home.jsx [hunk 1]
```diff
@@ -1,11 +1,16 @@
 import './Home.css';
 
-export function Home() {
+export function Home(props) {
+	const { onClick } = props;
```
Comment (itsoliviasparks) [inline src/views/Home.jsx]: If we wanted, we would destructure the props in the function declaration:

`export function Home({ onClick }) {};`

**Comment by itsoliviasparks** [comment]
Really nice work!

**Review by itsoliviasparks — APPROVED** [review]

File: src/App.jsx [hunk 2]
```diff
@@ -19,10 +26,15 @@ export function App() {
 	 * to create and join a new list.
 	 */
 	const [listToken, setListToken] = useStateWithStorage(
-		'my test list',
+		null,
 		'tcl-shopping-list-token',
 	);
 
+	const handleClick = useCallback(() => {
```
Comment (timothy-taylor) [inline src/App.jsx:33]: I think this useCallback is on the edge of premature optimization, but, still, it's cool to see you thinking about maintaining stable references to event handler functions.

File: src/App.jsx [hunk 3]
```diff
@@ -51,7 +63,16 @@ export function App() {
 		<Router>
 			<Routes>
 				<Route path="/" element={<Layout />}>
-					<Route index element={<Home />} />
+					<Route
+						index
+						element={
+							listToken ? (
+								<Navigate to="/list" />
```
Comment (timothy-taylor) [inline src/App.jsx:70]: Good call using the Navigate component and handling this at the Router level.

**Review by timothy-taylor — APPROVED** [review]
Nice work this week!

File: src/views/Home.jsx (hunk 1 above)
Comment (hansontram) [reply src/views/Home.jsx]: Thank you for this!

**Pull request merged**